Rename `floor_div` to `floor_divide` to match NumPy's API

sudarsan2k5 opened the pull request:
Addressing #1212 

- Replaced `floor_div` with `floor_divide`
- Deprecated `int_div`

+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [x] There are tests covering the changes introduced in the PR.

brandonwillard commented on aesara/tensor/math.py:
```diff
@@ -1771,8 +1771,8 @@ def minimum(x, y):
 def divmod(x, y):
-    """elementvise divmod, using floor_div and mod_check"""
-    return floor_div(x, y), mod_check(x, y)
+    """elementvise divmod, using floor_divide and mod_check"""
```
```suggestion
    """Element-wise `divmod`, using `floor_divide` and `mod_check`."""
```

sudarsan2k5 commented:
I believe there is still some `int_div` that I need to replace with `floor_div` and have to add the test for deprecation warning so converting this to draft.

sudarsan2k5 commented:
Hi @brandonwillard,

I made the below changes
1. replaced `floor_div` with `floor_divide`
2. added an alias `int_div` to `floor_divide`
3. removed some `int_div` imports and used `floor_divide` inplace of it
4. changed some function names and docs to maintain consistency across the code base
5. renamed `true_div_inplace` to `true_divide_inplace` for consistency
6. added tests for the deprecation of `int_div` and `floor_div`

sudarsan2k5 commented on tests/tensor/test_math.py:
```diff
@@ -3557,3 +3557,15 @@ def test_deprecations():
+    with pytest.deprecated_call():
+        from aesara.tensor.math import floor_div  # noqa: F401 F811
+
+    with pytest.deprecated_call():
+        from aesara.tensor import int_div  # noqa: F401 F811
+
+    with pytest.deprecated_call():
+        from aesara.tensor.math import int_div  # noqa: F401 F811
```
I have a question here: the test for deprecation of `floor_div` is working fine, but the test for `int_div` is not working; could you please point out where I went wrong?

rlouf replied:
Try to rebase your branch on `main`, I think this was fixed in https://github.com/aesara-devs/aesara/pull/1434

sudarsan2k5 replied:
I have already tried this, you can the tests which I have added for checking `floor_div` is working fine but the tests for `int_div` is not working, don't know what is the issue here, they both present at the same location.

brandonwillard commented on aesara/scalar/basic.py:
```diff
@@ -4645,7 +4650,7 @@ def __getattr__(name):
     "true_divide",
     "divide",
     "int_div",
```
```suggestion
```

brandonwillard commented on aesara/scalar/basic.py:
```diff
@@ -2133,10 +2133,9 @@ def grad(self, inputs, g_output):
 
-
-floor_div = int_div
+int_div = floor_divide
```
```suggestion
```

brandonwillard commented on tests/scalar/test_basic.py:
```diff
@@ -547,3 +547,15 @@ def test_deprecations():
+        from aesara.scalar import floor_div  # noqa: F401 F811
+
+    with pytest.deprecated_call():
+        from aesara.scalar.basic import int_div  # noqa: F401 F811
```
This is the one that's failing in CI.

brandonwillard commented on aesara/tensor/math.py:
```diff
@@ -1800,13 +1800,13 @@ def true_divide(a, b):
 
-# floor_div and int_div are the same thing
-floor_div = int_div
+# floor_divide and int_div are the same thing
+int_div = floor_divide
```
```suggestion
```
I don't think these references are necessary when an entry for the deprecated object is added to `DEPRECATED_NAMES` below.

brandonwillard commented on aesara/tensor/math.py:
```diff
@@ -3123,7 +3123,7 @@ def matmul(x1: "ArrayLike", x2: "ArrayLike", dtype: Optional["DTypeLike"] = None
     "mul",
     "true_divide",
     "int_div",
```
```suggestion
```

sudarsan2k5 commented:
![image](https://user-images.githubusercontent.com/108286640/219882715-617c8bc2-2e72-46cc-9a86-ef7eee7f63bb.png)

Thank you, Now this is working fine

sudarsan2k5 commented on aesara/scalar/basic.py:
```diff
@@ -2133,10 +2133,7 @@ def grad(self, inputs, g_output):
-
-
-floor_div = int_div
+floor_divide = IntDiv(upcast_out, name="floor_divide")
```
Since we are renaming everything, should we also rename the class `IntDiv`, I thing that will make our code consistence

rlouf replied:
It's a good idea

sudarsan2k5 commented:
Hi @rlouf I have renamed `floor_div` for `floor_divide`, added related tests and also renamed to related class name, could you please check if this PR is ready to merge or anything else is still pending? Thanks

rlouf commented:
Looks good to me. Thank you for contributing @sudarsan2k5, NumPy compatibility is very important.
